magento/magento2
# magento/magento2#22856: Catalog price rules are not working with custom options as expected.

### Description (*)


### Fixed Issues (if relevant)

1. magento/magento2#22856: Catalog price rules are not working with custom options as expected.

### Manual testing scenarios (*)

1. Create Product1 with fixed price custom options.
2. Create Product2 with percent price custom options.
3. Create and apply CatalogPriceRule1 with percent type discount.
4. Open Product1 and Product2 pages on the storefront and make sure discount is correctly applied to products options.
5. Disable CatalogPriceRule1.
6. Create and apply CatalogPriceRule2 with fixed type discount.
7. Open Product1 and Product2 pages on the storefront and make sure discount is correctly applied to products options.

### Contribution checklist (*)
